lib/realtime/tenants/connect.ex

```diff
@@ -103,6 +104,23 @@ defmodule Realtime.Tenants.Connect do
     end
   end
 
+  defp process_otel([]) do
+    span_ctx = Tracer.start_span("database.connect")
+    ctx = OpenTelemetry.Ctx.get_current()
+    OpenTelemetry.Ctx.attach(ctx)
+    Tracer.set_current_span(span_ctx)
+  end
+
+  defp process_otel(%{span_ctx: span_ctx, ctx: ctx}) do
+    OpenTelemetry.Ctx.attach(ctx)
+    Tracer.set_current_span(span_ctx)
+  end
+
+  defp process_otel(propagation) do
+    :otel_propagator_text_map.extract(propagation)
+    process_otel([])
+  end
```

Comment:
are we able to move this outside of the Connect code somehow? Connect it's already quite complex and this adds quite a lot of readability load to this code